File: benchmarks/cpp/recursive.cpp
// BENCH_RECURSIVE_DEPTH=20
#include <iostream>

static int recurse(int depth) {
    if (depth == 0) return 1;
    return recurse(depth - 1) + recurse(depth - 1);
}

int main() {
    std::cout << recurse(20) << '\n';
    return 0;
}

File: benchmarks/python/recursive.py
#!/usr/bin/env python3
"""BENCH_RECURSIVE_DEPTH=20"""


def recurse(depth: int) -> int:
    if depth == 0:
        return 1
    return recurse(depth - 1) + recurse(depth - 1)


def main() -> None:
    print(recurse(20))


if __name__ == "__main__":
    main()
